### PR TITLE
feat(kafka update): add reauthentication flag

### PR DESCRIPTION
**Pull request by rkpattnaik780**
Add `--reauthentication` flag to Kafka update

### Verification Steps

1. Set value for "enable_reauthentication" using the `--reauthentication` flag
`rhoas kafka update --reauthentication false`
2. Try running kafka update in interactive mode

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation change
- [ ] Other (please specify)

### Checklist

- [X] Documentation added for the feature
- [ ] CI and all relevant tests are passing
- [ ] Code Review completed
- [ ] Verified independently by reviewer

### CODE REVIEW

File: pkg/cmd/kafka/update/update.go
Comment (wtrocki) [inline pkg/cmd/kafka/update/update.go:130]: Ideally this should be removed before we merge.

File: pkg/cmd/kafka/update/update.go
Comment (wtrocki) [inline pkg/cmd/kafka/update/update.go:177]: This logic missing else and `"kafka.update.log.info.sameOwnerNoChanges` that were there previously. 
When adding flags we can eather:

- Make command execute only single operation per flag (owner, reauth)
- Make command execute multiple actions at the same time - will require priorities and decision how command should fail - Aggregation of errors that we done for acl etc.

File: pkg/cmd/kafka/update/update.go
Comment (wtrocki) [reply pkg/cmd/kafka/update/update.go:120]: Although vocabulary is not my strong point our flags should probably stick with nouns.
Also auth as shortcut doesn't carry that much info so my take is to drop enable and extend it to the full sentence.

reauthentication=true

File: pkg/cmd/kafka/update/update.go
Comment (rkpattnaik780) [reply pkg/cmd/kafka/update/update.go:177]: > will require priorities and decision how command should fail - Aggregation of errors that we done for acl etc.
I think this is something backend takes care of already, running a `kafka update --owner inexistent_user --reauthentication false` fails with error `user doesn't belong to your organization` and doesn't update the enable_reauthtentication field

Comment (wtrocki) [reply pkg/cmd/kafka/update/update.go:177]: Yes, but what is the plan to handle this in the CLI?

Do we want to even send request if we know that it will be wrong?
Do we provide info to user that error in owner also discarded other update (that might be assumed, but better to be explicit)

Comment (rkpattnaik780) [reply pkg/cmd/kafka/update/update.go:177]: > Do we want to even send request if we know that it will be wrong?

While we can check by fetching the list of users before sending request for update. I think we should let the backend check it for us.

> Do we provide info to user that error in owner also discarded other update

I would prefer not to be explicit about the discard as update sounds very much atomic. I will let you take the final call on it, maybe we could display a simple message like `update aborted`.

Comment (wtrocki) [reply pkg/cmd/kafka/update/update.go:177]: Please use your jugement based on my review. I just point out that relying on server side (especially server doesn't provide reliable error codes for that case that will help us to provide proper message to user) is bad UX. 

See entire terms and conditions check - backend does that but we still check that before :)

Comment (rkpattnaik780) [reply pkg/cmd/kafka/update/update.go:187]: current preview:
Screenshot from 2021-12-02 13-20-22

File: pkg/localize/locales/en/cmd/kafka_update.en.toml
Comment (wtrocki) [inline pkg/localize/locales/en/cmd/kafka_update.en.toml:53]: This looks different than what doc had :)
Should I correct this one?

Comment (rkpattnaik780) [reply pkg/localize/locales/en/cmd/kafka_update.en.toml:53]: The reviewed para looked too long, tried to combine  best bits from that and the other para without losing critical info.

Comment (wtrocki) [reply pkg/localize/locales/en/cmd/kafka_update.en.toml:53]: It lost meaning and form IMHO.
We need to inform user about side effect - 
this one has number of issues etc.

File: pkg/cmd/kafka/update/update.go
Comment (wtrocki) [reply pkg/cmd/kafka/update/update.go:140]: Lets remove that

File: pkg/cmd/kafka/update/update.go
Comment (wtrocki) [inline pkg/cmd/kafka/update/update.go:251]: Redundant. We usually want to return error with that message.

File: pkg/cmd/kafka/update/update.go
Comment (wtrocki) [inline pkg/cmd/kafka/update/update.go:253]: Use message here ^^

File: pkg/localize/locales/en/cmd/kafka_update.en.toml
Comment (wtrocki) [inline pkg/localize/locales/en/cmd/kafka_update.en.toml:24]: mixing `=` and ` ` in arguments

File: pkg/localize/locales/en/cmd/kafka_update.en.toml
Comment (wtrocki) [inline pkg/localize/locales/en/cmd/kafka_update.en.toml:58]:
```suggestion
- Use ACLs to prevent unauthorized connections from performing any operation
- Create support case for RHOSAK product
```

File: pkg/cmd/kafka/update/update.go
Comment (wtrocki) [inline pkg/cmd/kafka/update/update.go:187]:
```suggestion
	if opts.owner != ""  {
	  if opts.owner == kafkaInstance.GetOwner() {
	    // warn user
	  }
```

File: pkg/cmd/kafka/update/update.go
Comment (wtrocki) [inline pkg/cmd/kafka/update/update.go:192]: Same as above.

File: pkg/cmd/kafka/update/update.go
Comment (wtrocki) [inline pkg/cmd/kafka/update/update.go:54]: Why?

File: pkg/localize/locales/en/cmd/kafka_update.en.toml
Comment (wtrocki) [inline pkg/localize/locales/en/cmd/kafka_update.en.toml:83]:
```suggestion
one = "Enable connection reauthentication"
```

Comment (wtrocki) [reply pkg/localize/locales/en/cmd/kafka_update.en.toml:83]: As every flag is optional unless validated

File: pkg/cmd/kafka/update/update.go
Comment (rkpattnaik780) [reply pkg/cmd/kafka/update/update.go:128]: Using boolvar would mean it should be initialised to a default bool value. If the user runs command to change the owner, there would be no way for CLI to know if the `--reauthentication` was intended to change.

File: pkg/localize/locales/en/cmd/kafka_update.en.toml
Comment (rkpattnaik780) [reply pkg/localize/locales/en/cmd/kafka_update.en.toml:83]: added the [optional] tag in line with `kafka topic update` interactive. Basically to imply that it could be left blank

File: pkg/cmd/kafka/update/update.go
Comment (wtrocki) [reply pkg/cmd/kafka/update/update.go:128]: Yes. I hit that problem many times. We need tribool in CLI :D

File: pkg/localize/locales/en/cmd/kafka_update.en.toml
Comment (wtrocki) [reply pkg/localize/locales/en/cmd/kafka_update.en.toml:83]: Ah.. this is for interactive mode. Sorry I missed that mode. Please ignore.

File: pkg/cmd/kafka/update/update.go
Comment (rkpattnaik780) [reply pkg/cmd/kafka/update/update.go:140]: We need it as the run method fails with `Function 'run' has too many statements (58 > 50)`

File: pkg/cmd/kafka/update/update.go
Comment (rkpattnaik780) [reply pkg/cmd/kafka/update/update.go:187]: Not very much for of it as the existing update flows don't check for same values.

File: pkg/cmd/kafka/update/update.go
Comment (wtrocki) [reply pkg/cmd/kafka/update/update.go:140]: Can we make function have less statements :)

File: pkg/cmd/kafka/update/update.go
Comment (wtrocki) [reply pkg/cmd/kafka/update/update.go:187]: Not sure what you mean?

**Comment by wtrocki** [comment]
Is that still draft?

File: pkg/cmd/kafka/update/update.go
Comment (wtrocki) [inline pkg/cmd/kafka/update/update.go:53]: I would prefer if we declare tribool type in utils with those values rater than define something like this in the command

File: pkg/cmd/kafka/update/update.go
Comment (wtrocki) [reply pkg/cmd/kafka/update/update.go:139]: Not addressed

Comment (rkpattnaik780) [reply pkg/cmd/kafka/update/update.go:139]: The method has 120+ lines of code, same goes for multiple places where we have excluded lints.

Comment (wtrocki) [reply pkg/cmd/kafka/update/update.go:139]: Not sure what you mean. Do you mean that we do not want to address that line? or suggestion is to increase limit?

File: pkg/localize/locales/en/cmd/kafka_update.en.toml
Comment (wtrocki) [inline pkg/localize/locales/en/cmd/kafka_update.en.toml:53]: This change can affect your Kafka security
...If a bad actor

File: pkg/localize/locales/en/cmd/kafka_update.en.toml
Comment (wtrocki) [inline pkg/localize/locales/en/cmd/kafka_update.en.toml:57]: When `-` is used both markdown and ascidocs rendering doesn't provide values for that properly. 
We should use `*` across entire cli for lists. 

Note: let's create seprate issue

**Comment by wtrocki** [comment]
Verified, Reviewed

File: pkg/cmd/kafka/update/update.go
Comment (rkpattnaik780) [reply pkg/cmd/kafka/update/update.go:129]: I think this should be addressed in a separate issue, funlen check is skipped at several places and the the lines in this method is more than twice the limit.

File: pkg/cmdutil/flagutil/tribool.go
Comment (wtrocki) [inline pkg/cmdutil/flagutil/tribool.go:27]: Missing i18n

File: pkg/cmd/kafka/update/update.go
Comment (wtrocki) [reply pkg/cmd/kafka/update/update.go:129]: I fully understyand what you are saying but that is new place introduced in that specific PR. Using that logic every of review comment can be addressed as separate github issue. I'm really giving up on this.

File: pkg/cmdutil/flagutil/tribool.go
Comment (wtrocki) [inline pkg/cmdutil/flagutil/tribool.go:30]: Missing docs. 

Inspiration: https://github.com/grignaak/tribool/blob/master/tribool.go

File: pkg/cmdutil/flagutil/tribool.go
Comment (wtrocki) [inline pkg/cmdutil/flagutil/tribool.go:21]: I do thing this should be public method

**Review by wtrocki — APPROVED** [review]

**Comment by wtrocki** [comment]
The most polished PR I have seen for CLI.
Absolute masterpiece! 

Good to merge. Reviewed and verified!

**Review by wtrocki — APPROVED** [review]

File: pkg/cmd/kafka/update/update.go
Comment (craicoverflow) [inline pkg/cmd/kafka/update/update.go:340]: This would be a great helper as there a few commands which repeat that logic.

**Pull request merged**
